src/app/icd/ICDManager.cpp

```diff
@@ -15,12 +15,152 @@
  *    limitations under the License.
  */
 
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app-common/zap-generated/ids/Attributes.h>
+#include <app-common/zap-generated/ids/Clusters.h>
 #include <app/icd/ICDManager.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <platform/ConnectivityManager.h>
+#include <platform/LockTracker.h>
+#include <platform/internal/CHIPDeviceLayerInternal.h>
 
 namespace chip {
 namespace app {
 
-ICDManager::ICDManager() {}
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::IcdManagement;
 
+void ICDManager::ICDManager::Init()
+{
+    uint32_t activeModeInterval;
+    IcdManagement::Attributes::ActiveModeInterval::Get(kRootEndpointId, &activeModeInterval);
+    VerifyOrDie(kFastPollingInterval.count() < activeModeInterval);
+    UpdateIcdMode();
+    UpdateOperationStates(ActiveMode);
+}
+
+bool ICDManager::SupportCheckInProtocol()
+{
+    bool success;
+    uint32_t featureMap;
+    success = (IcdManagement::Attributes::FeatureMap::Get(kRootEndpointId, &featureMap) == EMBER_ZCL_STATUS_SUCCESS);
+
+    return success ? ((featureMap & to_underlying(IcdManagement::Feature::kCheckInProtocolSupport)) != 0) : false;
```

Comment:
can the feature map change at runtime? might be better if it as an attribute of the class instead.

Reply:
Featuremap is a read only attribute, It cannot change at run time if I am not mistaken.